### DISCUSSION TITLE
Digitization of receipts via OCR

### DISCUSSION THREAD

**Discussion by ambein** [mtotschnig/MyExpenses#1498] [Ideas]
Thanks for developing this app and providing a trial period with a commitment to opensource code.

I am interested in digitization of receipts for auto-population of expense reports via table structured OCR as well as a simple export feature for reporting and claim purposes (reimbursement requests, taxes, audits).

The OCR as currently implement is limited at the moment to mainly finding the total amount (most of the time). I have not found that there is recognition of payee name and dates/time. As mentioned in [previous issue](https://github.com/mtotschnig/MyExpenses/issues/1259#issue-1883594097) that may be possible. There are commercial attempts to implement this but I do not know how successful, as I did a [test here (with my own receipt) and the output was unstructured](https://ocr.space/receiptscanning). It would be amazing to develop and keep that technology in open source domain.

Now that some receipts can come in via email, a feature to read incoming receipts in multiple formats (e.g. via an email address or tagged message) might be an interesting feature.

With respect to export/report feature, the ability to email a report, source table, and image files would be great. The option to generate a print ready PDF report that is formatted with a summary/index page(s) and then cross referenced receipts, efficiently arranged on subsequent pages would be amazing. I.e. the report would include scanned receipts named and ordered in an organized fashion with summary index/ financial statement. Adding the source data for export (csv table, logically names jpgs of receipts, and the PDF report) would be nice.

**Comment by mtotschnig** [comment]
Actually, we currently try to scan for date/time and also compare text found in the receipt with the payee entries stored in the app.
Apart from that, the app has evolved historically by incrementally adding smaller features and supporting new use cases, so if you could describe a specific use case in detail, it would be easier to figure out its feasibility. 
I am aware that the current PDF print functionality can be improved with respect to receipts (#1479)
